scripts/functions/checksum

```diff
@@ -201,17 +201,27 @@ __rvm_checksum_read()
     [[ -n "${_checksum_md5:-}" ]] ||
       _checksum_md5="$(    __rvm_db_ "$rvm_user_path/md5"      "$_name" | \command \head -n 1 )"
 
-    [[ -n "${_checksum_md5:-}" && $_name == http* ]] ||
-       _checksum_md5="$(__rvm_curl -s -L $_name.md5)"
-
     # sha512
     _checksum_sha512="$(   __rvm_db_ "$rvm_path/config/sha512" "$_name" | \command \head -n 1 )"
 
     [[ -n "${_checksum_sha512:-}" ]] ||
       _checksum_sha512="$( __rvm_db_ "$rvm_user_path/sha512"   "$_name" | \command \head -n 1 )"
 
-    [[ -n "${_checksum_sha512:-}" && $_name == http* ]] ||
-       _checksum_sha512="$(__rvm_curl -s -L $_name.sha512)"
+    __rvm_checksum_none || return 0
+  done
+
+  # Only try to get the checksum from the network after checking the config files
```

Comment:
Why don't we keep the previous version but do:

```
[[ -z "${_checksum_md5:-}" && $_name == http* ]] ||		
        _checksum_md5="$(__rvm_curl -s -L $_name.md5)"
```

Reply:
I think that's incorrect too, it would run curl if `$_name` does not start with `http`, even if the checksum is already set. My first commit e370ad601fc045f3c886ad84fe2b04fa75af6eef I think does the minimal fix.

But it seems highly desirable to avoid `curl` calls entirely if there are already entries in the config files. Since the code iterates over multiple `_name`, we need to split the loop, so e.g. config files for the second `_name` (e.g., `truffleruby-19.0.0-linux-amd64.tar.gz`) are checked before a URL for the first `_name` (e.g., a URL). I think this approach somewhat concurs with @mpapis's comment: https://github.com/rvm/rvm/pull/4650#discussion_r269320930, i.e., avoiding needless `curl` calls.

I also find a pair of `if` a lot more readable and easier to reason about than multiple `&&`/`||` with side effects.